Automattic/wp-calypso
# Plugins: Reduxify plugin notice (AKA status) removal

Background: In the flux implementation, plugin statuses are stored in the "log-store", and are referenced as "notices" everywhere. In the new Redux implementation, these are called statuses and are referenced as statuses too.

This PR introduces a Redux action creator and reducer logic for removing plugin statuses by a status type (`completed`, `error`, or `inProgress`). It builds on top of what was already built for the installed plugin status Redux functionality. It also adds a compatibility layer to still dispatch the corresponding flux action, because we're still using the `log-store`. It will eventually be removed once we get rid of all `log-store` usages.

Furthermore, it updates all of the flux `removePluginsNotices` invocations to use Redux's `removePluginStatuses`. 

Finally, since this is the last plugins flux action (🎉 ), this PR removes the plugins flux actions file.

Part of #24180.

#### Changes proposed in this Pull Request

* State: Introduce Redux action creator and reducer for plugin statuses removal
* Plugins: Use Redux for removing plugin statuses
* Plugins: Remove unused flux actions

#### Testing instructions

* Get a Jetpack site with some plugins for testing.
* Compare against production and verify that notices still work the same way (and don't overlap each other, meaning, a success notice of the same type should previously remove the in progress notice of the same type) when updating, installing, or removing plugins, as well as toggling auto-updates and enabling/disabling plugins:
  * `/plugins/:plugin/:site`
  * `/plugins/:plugin`
  * `/plugins`
  * `/plugins/manage/:site` - for a single plugin
  * `/plugins/manage/:site` - for multiple selected plugins (bulk actions are enabled by clicking the "Edit All" button)
* Verify all tests pass.

#### Notes

Linting errors are expected - there are a couple of them in a file we touched. I'm not fixing them here to avoid making the PR extra large.
